backend/apps/mentorship/management/commands/link_issues_by_label.py

```diff
@@ -0,0 +1,63 @@
+"""A command to link issues to mentorship modules based on matching labels."""
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from apps.github.models.issue import Issue
+from apps.mentorship.models.module import Module
+
+
+class Command(BaseCommand):
+    """Efficiently link issues to mentorship modules based on matching labels."""
+
+    help = "Efficiently link issues to mentorship modules based on matching labels."
+
+    def handle(self, *_args, **options) -> None:
+        """Handle the command execution.
+
+        Args:
+            *_args: Variable length argument list.
+            **options: Arbitrary keyword arguments containing command options.
+
+        """
+        self.stdout.write("Fetching all issues and labels...")
+        # load id
+        issues = Issue.objects.prefetch_related("labels").only("id")
+
+        label_to_issue_ids: dict[str, set[int]] = {}
+        for issue in issues:
+            for label in issue.labels.all():
+                label_to_issue_ids.setdefault(label.name, set()).add(issue.id)
+
+        total_modules = 0
+        total_links = 0
+
+        self.stdout.write("Processing modules...")
+        # Filter out modules without linked_issue_labels upfront
+        modules = Module.objects.exclude(labels__isnull=True).exclude(labels=[])
+
+        for module in modules:
+            linked_labels = module.labels
+
+            matched_issue_ids = set()
+            for label in linked_labels:
+                matched_issue_ids.update(label_to_issue_ids.get(label, set()))
+
+            if matched_issue_ids:
+                current_ids = set(module.issues.values_list("id", flat=True))
+                if current_ids != matched_issue_ids:
+                    with transaction.atomic():
+                        module.issues.set(matched_issue_ids)
```

Comment:
You link any issue that has a label w/ this name -- we need project related issues only. The database contains issues from multiple repositories of different organizations.